config/routes.js

```diff
@@ -1,32 +1,69 @@
-const axios = require('axios');
+// require("dotenv").config();
 
-const { authenticate } = require('../auth/authenticate');
+const axios = require("axios");
+const db = require("../database/dbConfig");
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+const jwtSecret = process.env.JWT_SECRET;
+const { authenticate } = require("../auth/authenticate");
 
 module.exports = server => {
-  server.post('/api/register', register);
-  server.post('/api/login', login);
-  server.get('/api/jokes', authenticate, getJokes);
+  server.post("/api/register", register);
+  server.post("/api/login", login);
+  server.get("/api/jokes", authenticate, getJokes);
 };
 
+// TOKEN maker
+function generateToken(user) {
+  const payload = {
+    subject: user.id,
+    username: user.username
+  };
+  const secret = jwtSecret;
+  const options = {
+    expiresIn: "1h"
+  };
+  return jwt.sign(payload, secret, options);
+}
+
 function register(req, res) {
   // implement user registration
+  const creds = req.body;
+  creds.password = bcrypt.hashSync(creds.password, 12);
+  db("users")
+    .insert(creds)
+    .then(id => res.status(201).json(id))
```

Comment:
I think this sequence could use a bit of error checking; i.e., before ever inserting, making sure that the username and the password meet length requirements, etc.

Reply:
You could also be safe and make sure that the `id`, which is actually an array (typically containing one id), has at least one element.